src/java.base/share/native/libjli/parse_manifest.c

```diff
@@ -418,7 +492,29 @@ find_file(int fd, zentry *entry, const char *file_name)
          */
         if ((size_t)CENNAM(p) == JLI_StrLen(file_name) &&
           memcmp((p + CENHDR), file_name, JLI_StrLen(file_name)) == 0) {
-            if (JLI_Lseek(fd, base_offset + CENOFF(p), SEEK_SET) < (jlong)0) {
+            jlong cenlen = CENLEN(p);
+            jlong censiz = CENSIZ(p);
+            jlong cenoff = CENOFF(p);
+            jlong cenext = CENEXT(p);
+            if ((cenlen == ZIP64_MAGICVAL
+                  || censiz == ZIP64_MAGICVAL
+                  || cenoff == ZIP64_MAGICVAL)
+                && cenext > 0) {
+              Byte *base = p + CENHDR + CENNAM(p);
```

Comment:
How about adding a comment describing the start of then extended fields calculation?

// The start of the extended fields = cen_header_start + cen_header_fixed_fix + file_name_length

Reply:
Done